python/samples/getting_started/azure_functions/05_multi_agent_orchestration_concurrency/function_app.py

```diff
@@ -63,14 +71,38 @@ def multi_agent_concurrent_orchestration(context: DurableOrchestrationContext):
     physicist_thread = physicist.get_new_thread()
     chemist_thread = chemist.get_new_thread()
 
-    physicist_task = physicist.run(messages=str(prompt), thread=physicist_thread)
-    chemist_task = chemist.run(messages=str(prompt), thread=chemist_thread)
+    # Create generators from agent.run() calls
+    physicist_gen = physicist.run(messages=str(prompt), thread=physicist_thread)
+    chemist_gen = chemist.run(messages=str(prompt), thread=chemist_thread)
+
+    # Advance each generator to extract the underlying Durable Task objects
+    physicist_task = next(physicist_gen)
+    chemist_task = next(chemist_gen)
+
+    # Execute both tasks concurrently using task_all
+    task_results = yield context.task_all([physicist_task, chemist_task])
+
+    # Complete the generator protocol by sending results back
+    # Each generator returns its final value via StopIteration.value
+    physicist_result = None
+    chemist_result = None
+
+    try:
+        physicist_gen.send(task_results[0])
```

Comment:
I've never seen this direct access of generators and `send` in durable Python apps before, and I don't think it's intended that we make this a normal part of the programming model. Why is this needed, and can it be replaced with something more conventional?

Reply:
I understand what the sample is doing, but I don't understand why we can't use normal generator syntax. This is not something durable customers need to do in Python today: https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-overview?tabs=in-process%2Cnodejs-v3%2Cv1-model&pivots=python#fan-in-out

Reply:
Rearchitected to use `TaskBase` instead of `Generators` to be consistent and simpler from a customer PoV.